FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -238,39 +248,59 @@ - (NSError *_Nullable)mapFetchReleasesError:(NSError *)error {
 }
 
 - (void)fetchNewLatestRelease:(FIRAppDistributionUpdateCheckCompletion)completion {
-  [FIRFADApiService
-      fetchReleasesWithCompletion:^(NSArray *_Nullable releases, NSError *_Nullable error) {
-        if (error) {
+  [FIRFADApiService fetchReleasesWithCompletion:^(NSArray *_Nullable releases,
+                                                  NSError *_Nullable error) {
+    if (error) {
+      dispatch_async(dispatch_get_main_queue(), ^{
+        completion(nil, [self mapFetchReleasesError:error]);
+      });
+      return;
+    }
+
+    for (NSDictionary *releaseDict in releases) {
+      if ([[releaseDict objectForKey:kLatestReleaseKey] boolValue]) {
+        FIRFADInfoLog(@"Tester API - found latest release in response.");
+        NSString *displayVersion = [releaseDict objectForKey:kDisplayVersionKey];
+        NSString *buildVersion = [releaseDict objectForKey:kBuildVersionKey];
+
+        FIRFADInfoLog(@"Checking if version matches");
+        if (![self isCurrentVersion:displayVersion buildVersion:buildVersion]) {
```

Comment:
readability: I wonder if this would be more readable if we extracted the two checks into separate functions - so that we could have a single if -- we can still add the logging in the individual methods, but then we can have something like 

```
if (![self isCurrentVersion:releaseDict] || ![self codeIsIdentical:releaseDict]){
    FIRAppDistributionRelease *release =
              [[FIRAppDistributionRelease alloc] initWithDictionary:releaseDict];
      dispatch_async(dispatch_get_main_queue(), ^{
            FIRFADInfoLog(@"Found new release with version: %@", [release displayVersion]);
            completion(release, nil);
        });
    return;
}
```